lib/tapioca/compilers/dsl/smart_properties.rb

```diff
@@ -0,0 +1,169 @@
+# typed: true
+# frozen_string_literal: true
+
+require "parlour"
+
+begin
+  require "smart_properties"
+rescue LoadError
+  # means SmartProperties is not installed,
+  # so let's not even define the generator.
+  return
```

Comment:
@rafaelfranca This felt super hacky to do but it also seems like the best guard against `smart_properties` not being installed in the target application. If we can't load the gem, then not even defining this generator class seems to be the easiest way to guard it.

However, I can't be sure if there is a better way or if this is frowned upon. Can you comment?

Reply:
I think this good way to solve this problem.